src/components/Reown/NanoContract/BaseNanoContractRequest.js

```diff
@@ -418,6 +422,19 @@ export const BaseNanoContractRequest = ({
     );
   }
 
+  if (notRegistered && hasNcRegisterFailed) {
```

Comment:
I had to move up the modal because it wasn't showing up. It was always getting to the condition below